modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/StoreQueryParser.java

```diff
@@ -186,34 +270,91 @@ private static StoreQueryRuntime constructStoreQueryRuntime(AggregationRuntime a
-                                                                        variableExpressionExecutors) {
+                                                                        variableExpressionExecutors,
+                                                                LockWrapper lockWrapper) {
+        MatchingMetaInfoHolder metaStreamInfoHolder = null;
+        AbstractDefinition inputDefinition = null;
         metaStreamEvent.setEventType(EventType.TABLE);
-        initMetaStreamEvent(metaStreamEvent, table.getTableDefinition());
-        MatchingMetaInfoHolder metaStreamInfoHolder = generateMatchingMetaInfoHolder(metaStreamEvent,
-                table.getTableDefinition());
-        CompiledCondition compiledCondition = table.compileCondition(onCondition, metaStreamInfoHolder,
-                siddhiAppContext, variableExpressionExecutors, tableMap, queryName);
+
+        StoreQueryRuntime storeQueryRuntime = null;
         if (table instanceof QueryableProcessor) {
+            initMetaStreamEvent(metaStreamEvent, table.getTableDefinition());
+            metaStreamInfoHolder = generateMatchingMetaInfoHolder(metaStreamEvent, table.getTableDefinition());
+            CompiledCondition compiledCondition = table.compileCondition(onCondition, metaStreamInfoHolder,
+                    siddhiAppContext, variableExpressionExecutors, tableMap, queryName);
             List<Attribute> expectedOutputAttributes = buildExpectedOutputAttributes(storeQuery, siddhiAppContext,
                     tableMap, queryName, metaPosition, metaStreamInfoHolder);
             CompiledSelection compiledSelection = ((QueryableProcessor) table).compileSelection(
                     storeQuery.getSelector(), expectedOutputAttributes, metaStreamInfoHolder, siddhiAppContext,
                     variableExpressionExecutors, tableMap, queryName);
-            SelectStoreQueryRuntime storeQueryRuntime = new SelectStoreQueryRuntime((QueryableProcessor) table,
-                    compiledCondition, compiledSelection, expectedOutputAttributes, queryName);
+
+            storeQueryRuntime = new SelectStoreQueryRuntime((QueryableProcessor) table, compiledCondition,
+                    compiledSelection, expectedOutputAttributes, queryName);
             QueryParserHelper.reduceMetaComplexEvent(metaStreamInfoHolder.getMetaStateEvent());
             QueryParserHelper.updateVariablePosition(metaStreamInfoHolder.getMetaStateEvent(),
                     variableExpressionExecutors);
             return storeQueryRuntime;
         } else {
-            FindStoreQueryRuntime storeQueryRuntime = new FindStoreQueryRuntime(table, compiledCondition, queryName,
-                    metaStreamEvent);
-            populateFindStoreQueryRuntime(storeQueryRuntime, metaStreamInfoHolder, storeQuery.getSelector(),
-                    variableExpressionExecutors, siddhiAppContext, tableMap, queryName, metaPosition);
+            if (storeQuery.isDeleteQuery()) {
```

Comment:
Use a enum and switch statement

Reply:
Fixed with e206a5089d27e5abbd49123fe002f2426eba9cd8